app/Models/Color.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+
+/**
+ * Class Color.
+ *
+ * @property int|null $r
+ * @property int|null $g
+ * @property int|null $b
+ */
+class Color extends Model
+{
+	use HasFactory;
+	protected $guarded = ['id'];
+
+	public function photo(): BelongsTo
+	{
+		return $this->belongsTo(Photo::class, 'photo_id', 'id');
+	}
+
+	public function is_main($query)
```

Comment:
Where is the method called at all? If it is dead code, we should drop it.

Secondly, the method does not do anything with the current object (i.e. `$this`). Hence it should be a static function at least.